functions/src/index.ts

```diff
@@ -54,3 +54,20 @@ export const getImageUploadToken = https.onCall(async (data, context) => {
 
     return { uploadToken }
 })
+
+export const deleteImagefromStorage = firestoreDB
+    .document('settings/{settingsID}')
+    .onUpdate(async (change, context) => {
+        if (
+            change.before.data().logo &&
+            change.after.data().logo != change.before.data().logo
```

Comment:
Potensiell bug her: Dersom man allerede har ein logo, og deretter lastar opp ein ny logo, så vil den _nye_ logoen bli sletta.

Sidan filene får samme navn (images/ + ID), så vil nye logoar overskrive gamle. Så det kan berre vere 1 logo per tavla om gangen. Men `.logo`-feltet kan endre seg, fordi URLen inkluderer ein `token` som er unik for kvar opplasting. Så dersom du lastar opp samme bildet to gongar, så vil det andre overskrive det første, og så vil denne funksjonen slette det siste.

Eg trur det kan vere tryggare å kun slette dersom `after` er undefined:
```
if (change.before.data().logo && !change.after.data().logo) {
```
```

Reply:
Ja, enig. Bra catch:)